generic/lshw.py

```diff
@@ -0,0 +1,177 @@
+        self.run_cmd("lshw -class memory")
+        self.run_cmd("lshw -class cpu")
+        self.run_cmd("lshw -class volume")
+        self.run_cmd("lshw -class network")
+        self.run_cmd("lshw -class power")
+        self.run_cmd("lshw -class generic")
+        self.run_cmd("lshw -class processor")
+        self.run_cmd("lshw -class bridge")
+        self.run_cmd("lshw -class multimedia")
+        self.run_cmd("lshw -class display")
+        self.run_cmd("lshw -class system")
+        self.run_cmd("lshw -class communication")
+        self.run_cmd("lshw -class bus")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute "
+                      "verification" % self.is_fail)
+
+    def test_lshw_verification(self):
+        """
+        compare the output of lshw with other tools
+        which produces similar info of hardware.
+        """
+        # verifying mac address
+        get_mac = process.system_output(
+            " ip link show %s " % self.active_interface)
+        mac = re.search(r'link\/\ether (.*) brd', get_mac).group(1)
+        lshw_out_mac = process.system_output("lshw")
+        if mac not in lshw_out_mac:
+            self.fail("lshw failed to show correct mac address")
+
+        # verify network
+        lshw_out_net = process.system_output("lshw -class network")
+        if self.active_interface not in lshw_out_net:
+            self.fail("lshw failed to show correct active network interface")
+
+    def test_gen_rep(self):
+        """
+        Lshw is capable of producing reports in html, xml and json formats.
+        """
+        self.run_cmd("lshw -xml")
+        self.run_cmd("lshw -html -class disk")
+        self.run_cmd("lshw -json")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to generate report"
+                      % self.is_fail)
+
+    def test_businfo(self):
+        """
+        Outputs the device list showing bus information,
+        detailing SCSI, USB, IDE and PCI addresses.
+        """
+        bus_info = process.run("lshw -businfo")
+        if bus_info.exit_status:
+            self.fail(" lshw  failed to execute lshw -businfo  ")
+
+        # verifying the bus info for active network
+        get_bus = process.system_output('lshw -businfo').splitlines()
+        for line in get_bus:
+            if self.active_interface in line:
+                get_bus_info_act_inter = line.split(' ')[0].split(':', 1)[1]
+
+        get_bus_info_lspci = process.system_output("lspci -v ")
+        if get_bus_info_act_inter not in get_bus_info_lspci:
+            self.fail("Verification of network bus info failed ")
+
+    def test_sanitize(self):
+        """
+        sanitize output(remove sensitive information like serial numbers,etc.)
+        """
+        out = []
+        out_lshw = subprocess.Popen("lshw", stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE, shell=True)
```

Comment:
can process.run will be used 

like 
`out_lshw = process.run("lshw",shell=True)`